demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c

```diff
@@ -713,25 +715,53 @@ CK_RV xProvisionDevice( CK_SESSION_HANDLE xSession,
     CK_RV xResult;
     CK_FUNCTION_LIST_PTR pxFunctionList;
     CK_OBJECT_HANDLE xObject = 0;
+    CK_BYTE * pxPkcsLabels[] =
+    {
+        ( CK_BYTE * ) pkcs11configLABEL_DEVICE_CERTIFICATE_FOR_TLS,
+        ( CK_BYTE * ) pkcs11configLABEL_DEVICE_PRIVATE_KEY_FOR_TLS,
+    };
+    CK_OBJECT_CLASS xClass[] =
+    {
+        CKO_CERTIFICATE,
+        CKO_PRIVATE_KEY,
+    };
+    uint32_t ulObjectIndex = 0;
 
     xResult = C_GetFunctionList( &pxFunctionList );
 
     #if ( pkcs11configIMPORT_PRIVATE_KEYS_SUPPORTED == 1 )
-        if( xResult == CKR_OK )
+
+        /* Attempt to clean-up old crypto objects, but only if private key import is
+         * supported by this application, and only if the caller has provided new
+         * objects to use instead. */
+        if( ( NULL != pxParams->pucClientCertificate ) &&
+            ( NULL != pxParams->pucClientPrivateKey ) )
         {
-            xResult = xProvisionCertificate( xSession,
-                                             pxParams->pucClientCertificate,
-                                             pxParams->ulClientCertificateLength,
-                                             ( uint8_t * ) pkcs11configLABEL_DEVICE_CERTIFICATE_FOR_TLS,
-                                             &xObject );
+            xDestroyDefaultCryptoObjects( xSession );
```

Comment:
Regarding JITP certificate, I think that it would be ok to try to delete it and then warn, but not hard fail, if the call doesn't succeed. Any implementation might define that object as read-only, for example. 

In the latter case, import would probably fail, too, which suggests what I think is the better behavior for conditionally deleting and importing all of these certificate types: (1) check if they exist, (2) export them if they do, (3) compare them to the new values, (4) only import if they're different (or missing).